### PR TITLE
Road to EAPI-8

### PR DESCRIPTION
**Pull request by a17r**

### CODE REVIEW

File: app-crypt/openpgp-keys-kde/openpgp-keys-kde-20210718.ebuild [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	https://keys.openpgp.org/vks/v1/by-fingerprint/D81C0CB38EB725EF6691C385BB463350D6EF31EF
+		-> heiko.becker@kde.org.key
+	https://keys.openpgp.org/vks/v1/by-fingerprint/2D1D5B0588357787DE9EE225EC94D18F7F05997E
+		-> jr@jriddell.org
+	https://keys.openpgp.org/vks/v1/by-fingerprint/0AAC775BB6437A8D9AF7A3ACFE0784117FBCE11D
+		-> bshah@kde.org
+	https://keys.openpgp.org/vks/v1/by-fingerprint/D07BD8662C56CB291B316EB2F5675605C74E02CF
+		-> davidedmundson@kde.org
+	https://keys.openpgp.org/vks/v1/by-fingerprint/1FA881591C26B276D7A5518EEAAF29B42A678C20
+		-> notmart@gmail.com
```
Comment (mgorny) [inline app-crypt/openpgp-keys-kde/openpgp-keys-kde-20210718.ebuild:20]:
```suggestion
	https://keys.openpgp.org/vks/v1/by-fingerprint/2D1D5B0588357787DE9EE225EC94D18F7F05997E
		-> jr@jriddell.org.key
	https://keys.openpgp.org/vks/v1/by-fingerprint/0AAC775BB6437A8D9AF7A3ACFE0784117FBCE11D
		-> bshah@kde.org.key
	https://keys.openpgp.org/vks/v1/by-fingerprint/D07BD8662C56CB291B316EB2F5675605C74E02CF
		-> davidedmundson@kde.org.key
	https://keys.openpgp.org/vks/v1/by-fingerprint/1FA881591C26B276D7A5518EEAAF29B42A678C20
		-> notmart@gmail.com.key
```

File: app-crypt/openpgp-keys-kde/openpgp-keys-kde-20210718.ebuild [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+"
+
+LICENSE="public-domain"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86"
+
+S="${WORKDIR}"
```
Comment (mgorny) [inline app-crypt/openpgp-keys-kde/openpgp-keys-kde-20210718.ebuild:27]: It's nicer to keep S below SRC_URI.

File: eclass/kde.org.eclass [hunk 3]
```diff
@@ -116,6 +116,21 @@ if [[ ${EAPI} == 8 ]] && [[ -n ${KDE_ORG_COMMIT} ]]; then
 fi
 export KDE_BUILD_TYPE
 
+# @ECLASS-VARIABLE: KDE_ORG_SIG
+# @INTERNAL
+# @DESCRIPTION:
+# By default, this is set to "false"; for supported packages (KDE Frameworks,
+# Plasma and Gear) it is automatically set to "true".
+KDE_ORG_SIG="false"
+if [[ ${EAPI} == 8 ]] && [[ ${KDE_BUILD_TYPE} == release ]]; then
```
Comment (mgorny) [inline eclass/kde.org.eclass:125]:
```suggestion
if [[ ${EAPI} != 7 && ${KDE_BUILD_TYPE} == release ]]; then
```

File: eclass/kde.org.eclass [hunk 4]
```diff
@@ -116,6 +116,21 @@ if [[ ${EAPI} == 8 ]] && [[ -n ${KDE_ORG_COMMIT} ]]; then
...
+	if [[ ${KDE_GEAR} == true ]] || [[ ${CATEGORY} == kde-plasma ]] || [[ ${CATEGORY} == kde-frameworks ]]; then
```
Comment (mgorny) [inline eclass/kde.org.eclass:126]:
```suggestion
	if [[ ${KDE_GEAR} == true || ${CATEGORY} == kde-plasma || ${CATEGORY} == kde-frameworks ]]; then
```

File: eclass/kde.org.eclass [hunk 5]
```diff
@@ -116,6 +116,21 @@ if [[ ${EAPI} == 8 ]] && [[ -n ${KDE_ORG_COMMIT} ]]; then
...
+		KDE_ORG_SIG="true"
+		inherit verify-sig
+		VERIFY_SIG_OPENPGP_KEY_PATH="/usr/share/openpgp-keys/kde.org.asc"
```
Comment (mgorny) [inline eclass/kde.org.eclass:129]: I know a lot of packages do this wrong (and I need to really fix the example):

```suggestion
		VERIFY_SIG_OPENPGP_KEY_PATH="${BROOT}/usr/share/openpgp-keys/kde.org.asc"
```

File: eclass/cmake.eclass [hunk 6]
```diff
@@ -28,12 +28,14 @@ if [[ -z ${_CMAKE_ECLASS} ]]; then
...
 
 # @ECLASS-VARIABLE: BUILD_DIR
+# @DEFAULT_UNSET
 # @DESCRIPTION:
 # Build directory where all cmake processed files should be generated.
 # For in-source build it's fixed to ${CMAKE_USE_DIR}.
 # For out-of-source build it can be overridden, by default it uses
-# ${WORKDIR}/${P}_build.
-: ${BUILD_DIR:=${WORKDIR}/${P}_build}
+# ${WORKDIR}/${P}_build (in EAPI-8: ${CMAKE_USE_DIR}_build).
```
Comment (mgorny) [inline eclass/cmake.eclass:36]: I think it's better to describe the situation for the newest EAPI first, then special cases for older EAPIs.

File: eclass/cmake.eclass [hunk 7]
```diff
@@ -127,7 +129,11 @@ case ${CMAKE_MAKEFILE_GENERATOR} in
 esac
 
 if [[ ${PN} != cmake ]]; then
-	BDEPEND+=" dev-util/cmake"
+	if [[ ${EAPI} == 7 ]]; then
+		BDEPEND+=" dev-util/cmake"
+	else
+		BDEPEND+=" >=dev-util/cmake-3.18"
```
Comment (mgorny) [inline eclass/cmake.eclass:135]: Is there any harm in raising the dep unconditionally? FWICS it's the only stable version we have now.

File: eclass/cmake.eclass [hunk 8]
```diff
@@ -264,14 +270,22 @@ cmake-utils_useno() { _cmake_banned_func "" "$@" ; }
 # @DESCRIPTION:
 # Determine using IN or OUT source build
 _cmake_check_build_dir() {
-	: ${CMAKE_USE_DIR:=${S}}
+	if [[ ${EAPI} == 7 ]]; then
+		: ${CMAKE_USE_DIR:=${S}}
+	else
+		: ${CMAKE_USE_DIR:=$(pwd)}
```
Comment (mgorny) [inline eclass/cmake.eclass:276]:
```suggestion
		: ${CMAKE_USE_DIR:=${PWD}}
```

**Review by mgorny — COMMENTED** [review]
Do you need me to make a commit/PR testing the CMAKE_BUILD_TYPE change?

File: eclass/cmake.eclass (hunk 7 above)
Comment (a17r) [reply eclass/cmake.eclass:135]: If no one complains re changing deps without revision, no harm I could think of.

**Comment by a17r** [comment]
> Do you need me to make a commit/PR testing the CMAKE_BUILD_TYPE change?

I will take any of commit, PR, git am, attachment in bug or mail.

Thanks for your review so far!

**Pull request merged**